common/build/eslint-config-fluid/flat.mts

```diff
@@ -0,0 +1,328 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+// ESLint 9 flat-config compatibility wrapper for existing eslintrc configs.
+// Consumers can import { recommended, strict, minimalDeprecated } from this module
+// and spread them into their eslint.config.js.
+
+import { FlatCompat } from "@eslint/eslintrc";
+import eslintJs from "@eslint/js";
+import type { Linter } from "eslint";
+import path from "node:path";
+import { fileURLToPath } from "node:url";
+
+type FlatConfigArray = Linter.Config[];
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+
+const compat = new FlatCompat({
+	baseDirectory: __dirname,
+	recommendedConfig: eslintJs.configs.recommended,
+	allConfig: eslintJs.configs.all,
+});
+
+// Global ignores for all configs.
+// In flat config, a config object with only `ignores` is treated as a global ignore pattern.
+// See: https://eslint.org/docs/latest/use/configure/configuration-files#globally-ignoring-files-with-ignores
+const globalIgnores = {
```

Comment:
Nit: Eslint docs suggest using the `globalIgnores()` helper for this.

Using `defineConfig()` like most of the Eslint examples would also be nice but seem more complicated here, and I'm not sure if it gives us any extra value if we're already using the correct types (e.g. `FlatConfigArray`) when defining our configs.

Reply:
My plan is to completely rewrite these as native flat configs once we are on 9. So I'll make a note of this for when we do that part.